### ISSUE TITLE
IAM permission Read/Write/Delete/OtherActions

### ISSUE THREAD

**Issue by reach4bawer** [Azure/azure-sdk-for-python#15576]
How can I get the fine grained permissions that are listed under the following view - 

![image](https://user-images.githubusercontent.com/52763317/100638911-92285500-3302-11eb-9738-51aefaa650e7.png)

I am trying to get this via the API using python SDK for various Azure services. Can you please guide me in the right direction in terms of documentations or working examples?

**Comment by mccoyp** [comment]
Hi! Thanks for the question. You can use the management library of a service to get this type of information -- here are some samples for using the management libraries of each service: https://github.com/Azure-Samples/azure-samples-python-management

For example, to see access policy permissions for a Key Vault, you'd want to use the [azure-mgmt-keyvault](https://docs.microsoft.com/en-us/python/api/azure-mgmt-keyvault/azure.mgmt.keyvault?view=azure-python) library. Here's a quick code snippet of how to list the access policies for a given vault:

```python
from azure.identity import DefaultAzureCredential
from azure.mgmt.keyvault import KeyVaultManagementClient

SUBSCRIPTION_ID = "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"  # replace with your Azure subscription ID

credential = DefaultAzureCredential()
client = KeyVaultManagementClient(credential, SUBSCRIPTION_ID)

vault = client.vaults.get("resource-group", "vault")  # replace with your resource group and vault names
access_policies = vault.properties.access_policies

for access_policy in access_policies:
    print("Object ID of principal: {}".format(access_policy.object_id))
    print("Permissions: {}\n".format(access_policy.permissions))
```

You should see an output similar to:
```
Object ID of principal: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
Permissions: {'additional_properties': {}, 'keys': ['Get', 'List', 'Update', 'Create', 'Import', 'Delete', 'Recover', 'Backup', 'Restore'], 'secrets': ['Get', 'List', 'Set', 'Delete', 'Recover', 'Backup', 'Restore'], 'certificates': ['Get', 'List', 'Update', 'Create', 'Import', 'Delete', 'Recover', 'Backup', 'Restore', 'ManageContacts', 'ManageIssuers', 'GetIssuers', 'ListIssuers', 'SetIssuers', 'DeleteIssuers'], 'storage': None}

Object ID of principal: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
Permissions: {'additional_properties': {}, 'keys': ['Get', 'List', 'Update', 'Create', 'Import', 'Delete', 'Recover', 'Backup', 'Restore'], 'secrets': ['Get', 'List', 'Set', 'Delete', 'Recover'], 'certificates': ['Get', 'List'], 'storage': None}
```

**Comment by reach4bawer** [comment]
This is only showing things for KeyVault, is there something that can show this for all the services?

**Comment by mccoyp** [comment]
You can find reference documentation for each service at docs.microsoft.com: https://docs.microsoft.com/en-us/python/api/overview/azure/?view=azure-python. Services with a management library will have a "Management" tab under their service dropdown in the left-side table of contents, as shown in the image below:

![ToC](https://user-images.githubusercontent.com/39780829/105525283-b9adf500-5c95-11eb-9314-34ceeda3e83c.PNG)

Most services have a management library analogous to Key Vault's, and they all follow the naming pattern of "azure-mgmt-[service]". For example: [azure-mgmt-appconfiguration](https://docs.microsoft.com/en-us/python/api/azure-mgmt-appconfiguration/azure.mgmt.appconfiguration?view=azure-python), [azure-mgmt-cognitiveservices](https://docs.microsoft.com/en-us/python/api/azure-mgmt-cognitiveservices/azure.mgmt.cognitiveservices?view=azure-python), and [azure-mgmt-hybridcompute](https://docs.microsoft.com/en-us/python/api/azure-mgmt-hybridcompute/azure.mgmt.hybridcompute?view=azure-python). The documentation can be a bit sparse at times, but I hope the usage example with Key Vault above will provide a template for using other management libraries.

**Comment by reach4bawer** [comment]
Awesome. Thank you @mccoyp similar to your example for KeyVault if I was trying to get the role assignments just in the context of Key Vault is there a method to do that?

**Comment by mccoyp** [comment]
You're welcome! I saw that you posted an [issue](https://github.com/Azure/azure-sdk-for-python/issues/16299) about getting role assignments a little while ago -- I'll respond to that thread and post a solution there.

**Comment by budaesandrei** [comment]
Hi. Not sure this is the right place to post this query but I am trying to build an integration on Snowflake so when a new file is added in a blob storage, Snowflake will get triggered to consume it. I've managed to build the template in python. I am missing a way to automate the Access Control (IAM) part. Snowflake is creating a service principal on our Tenant and so far I was able through the Portal to assign it the "Storage Queue Data Contributor" role within my storage account. Is there currently a way to do that automatically?
